File: sdks/dotnet/src/Resources/OrdersResource.cs
using System.Threading.Tasks;
using Puxbay.SDK.Models;

namespace Puxbay.SDK.Resources
{
    public class OrdersResource
    {
        private readonly PuxbayClient _client;

        public OrdersResource(PuxbayClient client)
        {
            _client = client;
        }

        public async Task<PaginatedResponse<Order>> ListAsync(int page = 1, int pageSize = 20)
        {
            return await _client.GetAsync<PaginatedResponse<Order>>($"orders/?page={page}&page_size={pageSize}");
        }

        public async Task<Order> GetAsync(string orderId)
        {
            return await _client.GetAsync<Order>($"orders/{orderId}/");
        }

        public async Task<Order> CreateAsync(Order order)
        {
            return await _client.PostAsync<Order>("orders/", order);
        }

        public async Task<Order> CancelAsync(string orderId)
        {
            await _client.PostAsync<object>($"orders/{orderId}/cancel/", null);
            return await GetAsync(orderId);
        }
    }
}

File: sdks/php/src/Resources/OrdersResource.php
<?php

namespace Puxbay\Resources;

use Puxbay\Models\Order;
use Puxbay\Models\PaginatedResponse;

class OrdersResource extends BaseResource
{
    public function list(int $page = 1, int $pageSize = 20): PaginatedResponse
    {
        $data = $this->get("orders/?page={$page}&page_size={$pageSize}");
        return PaginatedResponse::fromArray($data, fn($item) => Order::fromArray($item));
    }

    public function getOrder(string $orderId): Order
    {
        $data = $this->get("orders/{$orderId}/");
        return Order::fromArray($data);
    }

    public function create(array $orderData): Order
    {
        // Note: Creating orders might use a simpler schema than the full Order object
        $data = $this->post('orders/', $orderData);
        return Order::fromArray($data);
    }

    public function cancel(string $orderId): Order
    {
        $data = $this->post("orders/{$orderId}/cancel/");
        return Order::fromArray($data);
    }
}
